### PR TITLE
DrivAerML: T_max=38 + lr=4.8e-4 MSE-only (fine schedule probe above T_max=36)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

T_max=36 is the clear schedule optimum at lr=5e-4 (val=3.622%). At lr=4.8e-4, the optimal T_max may shift slightly. T_max=38 probes just above T_max=36 to see if a marginally longer cycle helps at the slightly lower LR.

Alphonse #3388 is testing T_max=40 + lr=4.8e-4. This fills the gap between T_max=36 and T_max=40.

## Instructions

```bash
cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml --optimizer adamw --lr 4.8e-4 \
  --cosine-t-max 38 --grad-clip 0.5 --enable-fourier \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --ema-decay 0.9995 --ema-mode fixed --no-compile-model --save-checkpoint \
  --wandb-group dm-lr48e4-tmax36 \
  --agent canute --wandb-name "canute/dm-lr48e4-tmax38"
```

MSE-only. If checkpoint saved, run full-eval with `--max-eval-batches 0 --load-checkpoint <path> --eval-only`.

## Baseline
- **DM best val:** 3.622% (T_max=36 MSE-only at lr=5e-4)
- **DM best TEST:** 4.117% (gojo lr=4.8e-4 T_max=30)
- **Target: val < 3.622%**

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT canute:

## Results

### Sampled eval (200 val batches, during training)
| Metric | Baseline | This run |
|--------|----------|----------|
| Val surface_rel_l2_pct | 3.622% | **3.705%** |
| Test surface_rel_l2_pct | 4.117% | 4.572% |

### Full eval (all batches, from best checkpoint epoch 817)
| Metric | Baseline | This run |
|--------|----------|----------|
| Val surface_rel_l2_pct | 3.622% | **4.617%** |
| Test surface_rel_l2_pct | 4.117% | **4.201%** |

### Summary
- **Best sampled val**: 3.705% at epoch 817 (vs target 3.622%) — did NOT beat val baseline
- **Full-eval val**: 4.617% — significantly worse than sampled val, indicating the 200-batch eval subset was overly optimistic
- **Full-eval test**: 4.201% — did NOT beat DM best TEST (4.117%)
- **Epochs trained**: 861 (hit 600-min timeout)
- **Peak VRAM**: ~7.4 GB / 97.9 GB (H200)

### Train command
```bash
cd target/icml2026 && SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml --optimizer adamw --lr 4.8e-4 \
  --cosine-t-max 38 --grad-clip 0.5 --enable-fourier \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --ema-decay 0.9995 --ema-mode fixed --no-compile-model --save-checkpoint \
  --wandb-group dm-lr48e4-tmax36 \
  --agent canute --wandb-name "canute/dm-lr48e4-tmax38"
```

### W&B runs
- Training: [t8xn82d9](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/t8xn82d9)
- Full-eval: [cz4fso50](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/cz4fso50)

### What happened
T_max=38 at lr=4.8e-4 converged slowly but steadily over 861 epochs (~10 hours). The sampled val (200 batches) appeared promising at 3.705%, tantalizingly close to the 3.622% baseline. However, the full-eval reveals the model was overfitting to the evaluation subset — full val is 4.617%, far above the baseline.

The longer cosine cycle (T_max=38 vs 36) did not improve over the T_max=36 baseline. The model reached its best sampled val around epoch 817 and appeared to be plateauing.

### Suggested follow-ups
- The large gap between sampled val (3.705%) and full val (4.617%) suggests that 200 eval batches may not be representative enough for DrivAerML. Consider increasing `--max-eval-batches` to improve signal quality.
- T_max=38 is slightly worse than T_max=36, consistent with T_max=36 being the local optimum for this LR range.
- The search could focus on other hyperparameters (e.g. weight decay, grad clip) rather than further T_max exploration above 36.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3403 because ADVISOR: Closing — T_max=38 + lr=4.8e-4 did not beat baseline. Full-eval val=4.617% and test=4.201% vs baselines of 3.622% and 4.117%. The large gap between sampled val (3.705%) and full-eval val (4.617%) also confirms the 200-batch eval subset is not representative. T_max=36 remains the schedule optimum. Thank you canute..
